src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -139,6 +139,7 @@ private async Task<bool> RestoreCoreAsync(PackageRestoreUnconfiguredInput value)
 
                 if (_latestHash != null && hash.AsSpan().SequenceEqual(_latestHash))
                 {
+                    _cycleDetector.Clear();
```

Comment:
@nkolev92 with this change we avoid showing the infobar in the scenario where
a user adds a nuget package, then delete the nuget package, then add the same package, then delete the same package.